# Задание 6
## Решение

In [55]:
import plotly.express as px
import pandas as pd

In [56]:
grad = pd.read_csv('6.csv', sep=';', decimal=',')
grad

College  %_grad_on_time
0        A            57.6
1        A            43.2
2        A            49.6
3        A            51.4
4        A            69.9
5        A            69.9
6        A            73.8
7        A            72.3
8        B            70.1
9        B            69.6
10       B            67.3
11       B            76.7
12       B            69.4
13       B            72.6
14       B            70.2
15       B            74.4
16       C            54.5
17       C            55.6
18       C            56.9
19       C            71.0
20       C            73.3
21       C            74.8
22       C            68.3
23       C            67.0
24       D            80.1
25       D            77.3
26       D            74.7
27       D            79.2
28       D            84.6
29       D            78.8
30       D            84.1
31       D            74.1
32       E            71.3
33       E            62.6
34       E            54.5
35       E            57.5
36       E            55.0
37       E            67.0
38       E            60.0
39       E            58.3
40       F            68.8
41       F            61.0
42       F            57.7
43       F            66.4
44       F            75.2
45       F            87.4
46       F            86.0
47       F            79.7

In [57]:
title = 'Данные о доле успешно завершивших обучение'
y_label = 'Процент окончивших'
x_label = 'Колледж'
labels = {'%_grad_on_time': y_label, 'College': x_label}

In [58]:
fig = px.box(grad, x='College', y='%_grad_on_time', color='College', points='all', title=title, labels=labels, width=500, height=400)

fig.show()

## Ответы на вопросы
### Вопрос 1
Я бы выбрал колледж `D`, там стабильно высокий процент оканчивает обучение, а также самый высокий средний процент окончивших

### Вопрос 2
Колледж `F` не стабильный, данные варьируются от уровня ниже 60% до уровня выше 85%. В то время как в колледже `B` данные распределены стабильно около 70%. __Я из Беларуси - я за стабильность. Я выберу колледж `B`.__